src/DocumentPicker.Sample/DocumentPickerPage.xaml.cs

```diff
@@ -2,60 +2,74 @@
+using FileSystem = Xamarin.Essentials.FileSystem;
+using Image = Xamarin.Forms.Image;
+using System.Data.SqlTypes;
+using Color = Svg.Interfaces.Color;
 
 namespace DocumentPicker.Samples
 {
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class DocumentPickerPage : ContentPage
     {
-        ObservableCollection<FileResult> files = new ObservableCollection<FileResult>();
+
+        private readonly DocumentConverterService _converterService = new DocumentConverterService();
+
+        ObservableCollection<DocumentResult> files = new ObservableCollection<DocumentResult>();
         public DocumentPickerPage()
         {
             InitializeComponent();
 
-            //files.CollectionChanged += Files_CollectionChanged;
+            files.CollectionChanged += Files_CollectionChanged;
 
             pickDocument.Clicked += async (sender, args) =>
             {
-                var x = new DocumentPickerService();
+                var fileResult = await _converterService.PickDocumentAsync();
 
-                var fileResult = await x.PickDocumentAsync();
-                
                 files.Add(fileResult);
+
             };
         }
 
-   //     private void Files_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
-   //     {
-   //         if (files.Count == 0)
-   //         {
-   //             ImageList.Children.Clear();
-   //             return;
-   //         }
-   //         if (e.NewItems.Count == 0)
-   //             return;
-
-   //         var file = e.NewItems[0] as DocumentFile;
-   //         var image = new Image { WidthRequest = 300, HeightRequest = 300, Aspect = Aspect.AspectFit };
-   //         if (file != null)
-   //         {
-   //             image.Source = ImageSource.FromFile(file.FileResult.FullPath);
-   //             /*image.Source = ImageSource.FromStream(() =>
-   //{
-   //    var stream = file.GetStream();
-   //    return stream;
-   //});*/
-   //             ImageList.Children.Add(image);
-
-   //             var image2 = new CachedImage { WidthRequest = 300, HeightRequest = 300, Aspect = Aspect.AspectFit };
-   //             image2.Source = ImageSource.FromFile(file.FileResult.FullPath);
-   //             ImageList.Children.Add(image2);
-   //         }
-   //     }
+        private void Files_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            if (files.Count == 0)
+            {
+                ImageList.Children.Clear();
+                return;
+            }
+            if (e.NewItems.Count == 0)
+                return;
+
+            var file = e.NewItems[0] as DocumentResult;
+
+            var svgString = _converterService.ConvertPdfToSvgString(file);
+            var svgDoc = SvgDocument.FromSvg<SvgDocument>(svgString);
+
+            var newPngFilePath = Path.Combine(FileSystem.CacheDirectory, file?.FileName.Split('.')[0] + ".png");
```

Comment:
Please use `System.IO.Path.GetFileNameWithoutExtension()` and `System.IO.Path.GetExtension()` instead!
Also, you do not need to get the path and the filename.
If you just have the full path, you can use `System.IO.Path.GetFilename()` to infer the filename